opentelemetry-api/src/opentelemetry/trace/__init__.py

```diff
@@ -273,6 +285,103 @@ class Tracer(ABC):
     and controlling spans' lifecycles.
     """
 
+    def decorate(
+        self,
+        name: str = None,
+        context: Optional[Context] = None,
+        kind: SpanKind = SpanKind.INTERNAL,
+        attributes: types.Attributes = None,
+        links: _Links = None,
+        start_time: Optional[int] = None,
+        record_exception: bool = True,
+        set_status_on_exception: bool = True,
+        end_on_exit: bool = True,
+        span_arg_name: str | None = "span",
+    ) -> Callable[[Callable[P, R]], Callable[P, R]]:
+        """Decorate the function with a span.
+
+        Decorate the function with a span. For more information see `start_as_current_span`.
+
+        Example::
+
+            @tracer.decorate()
+            def foo():
+                pass
+
+        Args:
+            name: The name of the span to be created, if not provided, the function
+                name will be used.
+            context: An optional Context containing the span's parent. Defaults to the
+                global context.
+            kind: The span's kind (relationship to parent). Note that is
+                meaningful even if there is no parent.
+            attributes: The span's attributes.
+            links: Links span to other spans
+            start_time: Sets the start time of a span
+            record_exception: Whether to record any exceptions raised within the
+                context as error event on the span.
+            set_status_on_exception: Only relevant if the returned span is used
+                in a with/context manager. Defines whether the span status will
+                be automatically set to ERROR when an uncaught exception is
+                raised in the span with block. The span status won't be set by
+                this mechanism if it was previously set manually.
+            end_on_exit: Whether to end the span automatically when leaving the
+                context manager.
+            span_arg_name: Lookup for this span in the function's kwargs to pass
+                the span to the function. If None, the span will not be passed
+                to the function.
+        """
+
+        def __decorator(func: Callable[P, R]) -> Callable[P, R]:
+            _span_name = name or func.__name__
+            if bool(func.__code__.co_flags & 0x80):
```

Comment:
Is there a reason to use this over `inspect.iscoroutinefunction()`?

Reply:
Oh, I didn't know this existed...
Pushing a fix